cmd/snap/cmd_advise.go

```diff
@@ -205,11 +211,39 @@ func adviseViaAptHook() error {
 	return nil
 }
 
+func dumpDbHook() error {
+	commands, err := advisor.DumpCommands()
+	if err != nil {
+		return err
+	}
+
+	commands_processed := make([]string, 0, len(commands))
+	re := regexp.MustCompile(`\A\[\{\"snap\"\:\"([^\"]*)\"`)
```

Comment:
I somehow missed this when reviewing. Why are you using regexps instead of parsing the json?

Reply:
Would you mind if I pushed a commit that changed it to use json?

Reply:
Augh, the more I look at this the worse it gets: this code completely ignores the fact that one command could be in more than one snap.

For example, today, there are already a couple of snaps that provide `rg` and `swift`:
```json
[{"Snap":"rg","Version":"0.8.1","Command":"rg"},{"Snap":"ripgrep","Version":"0.8.1","Command":"rg"}]
[{"Snap":"openstackclients","Version":"queens","Command":"swift"},{"Snap":"swift","Version":"4.0.3","Command":"swift"}]
```

how should those be printed for your use case?

Reply:
There should just be multiple entries with the same command name.